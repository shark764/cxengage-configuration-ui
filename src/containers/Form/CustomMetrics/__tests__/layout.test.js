/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';

import CustomMetricsForm from '../layout';

describe('<CustomMetricsForm />', () => {
  it('renders create customMetric', () => {
    const rendered = shallow(
      <CustomMetricsForm
        name="Custom metric"
        description="Custom metric description"
        customMetricsType="SLA"
        slaAbandonType="ignored-abandoned-calls"
        slaThreshold="20"
        slaAbandonThreshold="20"
        active={false}
        isSaving={false}
        inherited={false}
        userHasUpdatePermission={true}
        handleSubmit={() => {}}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders update customMetric', () => {
    const rendered = shallow(
      <CustomMetricsForm
        name="Custom metric"
        description="Custom metric description"
        customMetricsType="SLA"
        slaAbandonType="ignored-abandoned-calls"
        id="57083781-332d-11e6-8dd4-c88eee4d9f61"
        slaThreshold="20"
        slaAbandonThreshold="20"
        active={false}
        isSaving={false}
        inherited={false}
        userHasUpdatePermission={true}
        handleSubmit={() => {}}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders update customMetric with ignored-abandoned-calls type', () => {
    const rendered = shallow(
      <CustomMetricsForm
        name="Custom metric"
        description="Custom metric description"
        customMetricsType="SLA"
        slaAbandonType="ignored-abandoned-calls"
        id="57083781-332d-11e6-8dd4-c88eee4d9f61"
        slaThreshold="20"
        slaAbandonThreshold="20"
        active={false}
        isSaving={false}
        inherited={false}
        userHasUpdatePermission={true}
        handleSubmit={() => {}}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders update customMetric saving and inherited with ignored-abandoned-calls type', () => {
    const rendered = shallow(
      <CustomMetricsForm
        name="Custom metric"
        description="Custom metric description"
        customMetricsType="SLA"
        slaAbandonType="ignored-abandoned-calls"
        id="57083781-332d-11e6-8dd4-c88eee4d9f61"
        slaThreshold="20"
        slaAbandonThreshold="20"
        active={false}
        isSaving={true}
        inherited={true}
        userHasUpdatePermission={true}
        handleSubmit={() => {}}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders update customMetric with count-against-sla type', () => {
    const rendered = shallow(
      <CustomMetricsForm
        name="Custom metric"
        description="Custom metric description"
        customMetricsType="SLA"
        slaAbandonType="count-against-sla"
        id="57083781-332d-11e6-8dd4-c88eee4d9f61"
        slaThreshold="20"
        active={false}
        isSaving={false}
        inherited={false}
        userHasUpdatePermission={true}
        handleSubmit={() => {}}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders update customMetric saving and inherited with count-against-sla type', () => {
    const rendered = shallow(
      <CustomMetricsForm
        name="Custom metric"
        description="Custom metric description"
        customMetricsType="SLA"
        slaAbandonType="count-against-sla"
        id="57083781-332d-11e6-8dd4-c88eee4d9f61"
        slaThreshold="20"
        active={false}
        isSaving={true}
        inherited={true}
        userHasUpdatePermission={true}
        handleSubmit={() => {}}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders update customMetric without update permision', () => {
    const rendered = shallow(
      <CustomMetricsForm
        name="Custom metric"
        description="Custom metric description"
        customMetricsType="SLA"
        slaAbandonType="ignored-abandoned-calls"
        id="57083781-332d-11e6-8dd4-c88eee4d9f61"
        slaThreshold="20"
        slaAbandonThreshold="20"
        active={false}
        isSaving={false}
        inherited={false}
        userHasUpdatePermission={false}
        handleSubmit={() => {}}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
