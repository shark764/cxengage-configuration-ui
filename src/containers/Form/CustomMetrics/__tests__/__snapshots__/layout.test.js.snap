// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomMetricsForm /> renders create customMetric 1`] = `
<form
  onSubmit={[Function]}
>
  <DetailHeader
    text="Details"
  />
  <InputField
    componentType="input"
    disabled={false}
    id="frm-custom-metrics-name"
    inputType="text"
    label="Name *"
    name="name"
  />
  <InputField
    componentType="textarea"
    disabled={false}
    id="frm-custom-metrics-description"
    inputType="text"
    label="Description"
    name="description"
  />
  <DetailHeader
    text="Properties"
  />
  <InputField
    componentType="input"
    dataType="number"
    disabled={false}
    id="frm-custom-metrics-sla-threshold"
    inputType="text"
    label="SLA Threshold"
    labelHelpText="(seconds)"
    name="slaThreshold"
  />
  <RadioGroupField
    disabled={false}
    id="frm-custom-metrics-sla-abandon-type"
    label="SLA Abandon Type"
    name="slaAbandonType"
    options={
      Array [
        Object {
          "helpText": "(Filter Out / Exclude Short Abandons)",
          "label": "Ignore Abandons",
          "value": "ignored-abandoned-calls",
        },
        Object {
          "helpText": "(Include short abandons against/in SLA)",
          "label": "Count Against SLA",
          "value": "count-against-sla",
        },
      ]
    }
    required={true}
  />
  <InputField
    componentType="input"
    dataType="number"
    disabled={false}
    id="frm-custom-metrics-sla-abandon-threshold"
    inputType="text"
    label="Abandon Threshold"
    labelHelpText="(seconds)"
    name="slaAbandonThreshold"
  />
</form>
`;

exports[`<CustomMetricsForm /> renders update customMetric 1`] = `
<form
  onSubmit={[Function]}
>
  <DetailHeader
    text="Details"
  />
  <InputField
    componentType="input"
    disabled={false}
    id="frm-custom-metrics-name"
    inputType="text"
    label="Name *"
    name="name"
  />
  <InputField
    componentType="textarea"
    disabled={false}
    id="frm-custom-metrics-description"
    inputType="text"
    label="Description"
    name="description"
  />
  <DetailHeader
    text="Properties"
  />
  <InputField
    componentType="input"
    dataType="number"
    disabled={false}
    id="frm-custom-metrics-sla-threshold"
    inputType="text"
    label="SLA Threshold"
    labelHelpText="(seconds)"
    name="slaThreshold"
  />
  <RadioGroupField
    disabled={false}
    id="frm-custom-metrics-sla-abandon-type"
    label="SLA Abandon Type"
    name="slaAbandonType"
    options={
      Array [
        Object {
          "helpText": "(Filter Out / Exclude Short Abandons)",
          "label": "Ignore Abandons",
          "value": "ignored-abandoned-calls",
        },
        Object {
          "helpText": "(Include short abandons against/in SLA)",
          "label": "Count Against SLA",
          "value": "count-against-sla",
        },
      ]
    }
    required={true}
  />
  <InputField
    componentType="input"
    dataType="number"
    disabled={false}
    id="frm-custom-metrics-sla-abandon-threshold"
    inputType="text"
    label="Abandon Threshold"
    labelHelpText="(seconds)"
    name="slaAbandonThreshold"
  />
</form>
`;

exports[`<CustomMetricsForm /> renders update customMetric saving and inherited with count-against-sla type 1`] = `
<form
  onSubmit={[Function]}
>
  <DetailHeader
    text="Details"
  />
  <InputField
    componentType="input"
    disabled={true}
    id="frm-custom-metrics-name"
    inputType="text"
    label="Name *"
    name="name"
  />
  <InputField
    componentType="textarea"
    disabled={true}
    id="frm-custom-metrics-description"
    inputType="text"
    label="Description"
    name="description"
  />
  <DetailHeader
    text="Properties"
  />
  <InputField
    componentType="input"
    dataType="number"
    disabled={true}
    id="frm-custom-metrics-sla-threshold"
    inputType="text"
    label="SLA Threshold"
    labelHelpText="(seconds)"
    name="slaThreshold"
  />
  <RadioGroupField
    disabled={true}
    id="frm-custom-metrics-sla-abandon-type"
    label="SLA Abandon Type"
    name="slaAbandonType"
    options={
      Array [
        Object {
          "helpText": "(Filter Out / Exclude Short Abandons)",
          "label": "Ignore Abandons",
          "value": "ignored-abandoned-calls",
        },
        Object {
          "helpText": "(Include short abandons against/in SLA)",
          "label": "Count Against SLA",
          "value": "count-against-sla",
        },
      ]
    }
    required={true}
  />
</form>
`;

exports[`<CustomMetricsForm /> renders update customMetric saving and inherited with ignored-abandoned-calls type 1`] = `
<form
  onSubmit={[Function]}
>
  <DetailHeader
    text="Details"
  />
  <InputField
    componentType="input"
    disabled={true}
    id="frm-custom-metrics-name"
    inputType="text"
    label="Name *"
    name="name"
  />
  <InputField
    componentType="textarea"
    disabled={true}
    id="frm-custom-metrics-description"
    inputType="text"
    label="Description"
    name="description"
  />
  <DetailHeader
    text="Properties"
  />
  <InputField
    componentType="input"
    dataType="number"
    disabled={true}
    id="frm-custom-metrics-sla-threshold"
    inputType="text"
    label="SLA Threshold"
    labelHelpText="(seconds)"
    name="slaThreshold"
  />
  <RadioGroupField
    disabled={true}
    id="frm-custom-metrics-sla-abandon-type"
    label="SLA Abandon Type"
    name="slaAbandonType"
    options={
      Array [
        Object {
          "helpText": "(Filter Out / Exclude Short Abandons)",
          "label": "Ignore Abandons",
          "value": "ignored-abandoned-calls",
        },
        Object {
          "helpText": "(Include short abandons against/in SLA)",
          "label": "Count Against SLA",
          "value": "count-against-sla",
        },
      ]
    }
    required={true}
  />
  <InputField
    componentType="input"
    dataType="number"
    disabled={true}
    id="frm-custom-metrics-sla-abandon-threshold"
    inputType="text"
    label="Abandon Threshold"
    labelHelpText="(seconds)"
    name="slaAbandonThreshold"
  />
</form>
`;

exports[`<CustomMetricsForm /> renders update customMetric with count-against-sla type 1`] = `
<form
  onSubmit={[Function]}
>
  <DetailHeader
    text="Details"
  />
  <InputField
    componentType="input"
    disabled={false}
    id="frm-custom-metrics-name"
    inputType="text"
    label="Name *"
    name="name"
  />
  <InputField
    componentType="textarea"
    disabled={false}
    id="frm-custom-metrics-description"
    inputType="text"
    label="Description"
    name="description"
  />
  <DetailHeader
    text="Properties"
  />
  <InputField
    componentType="input"
    dataType="number"
    disabled={false}
    id="frm-custom-metrics-sla-threshold"
    inputType="text"
    label="SLA Threshold"
    labelHelpText="(seconds)"
    name="slaThreshold"
  />
  <RadioGroupField
    disabled={false}
    id="frm-custom-metrics-sla-abandon-type"
    label="SLA Abandon Type"
    name="slaAbandonType"
    options={
      Array [
        Object {
          "helpText": "(Filter Out / Exclude Short Abandons)",
          "label": "Ignore Abandons",
          "value": "ignored-abandoned-calls",
        },
        Object {
          "helpText": "(Include short abandons against/in SLA)",
          "label": "Count Against SLA",
          "value": "count-against-sla",
        },
      ]
    }
    required={true}
  />
</form>
`;

exports[`<CustomMetricsForm /> renders update customMetric with ignored-abandoned-calls type 1`] = `
<form
  onSubmit={[Function]}
>
  <DetailHeader
    text="Details"
  />
  <InputField
    componentType="input"
    disabled={false}
    id="frm-custom-metrics-name"
    inputType="text"
    label="Name *"
    name="name"
  />
  <InputField
    componentType="textarea"
    disabled={false}
    id="frm-custom-metrics-description"
    inputType="text"
    label="Description"
    name="description"
  />
  <DetailHeader
    text="Properties"
  />
  <InputField
    componentType="input"
    dataType="number"
    disabled={false}
    id="frm-custom-metrics-sla-threshold"
    inputType="text"
    label="SLA Threshold"
    labelHelpText="(seconds)"
    name="slaThreshold"
  />
  <RadioGroupField
    disabled={false}
    id="frm-custom-metrics-sla-abandon-type"
    label="SLA Abandon Type"
    name="slaAbandonType"
    options={
      Array [
        Object {
          "helpText": "(Filter Out / Exclude Short Abandons)",
          "label": "Ignore Abandons",
          "value": "ignored-abandoned-calls",
        },
        Object {
          "helpText": "(Include short abandons against/in SLA)",
          "label": "Count Against SLA",
          "value": "count-against-sla",
        },
      ]
    }
    required={true}
  />
  <InputField
    componentType="input"
    dataType="number"
    disabled={false}
    id="frm-custom-metrics-sla-abandon-threshold"
    inputType="text"
    label="Abandon Threshold"
    labelHelpText="(seconds)"
    name="slaAbandonThreshold"
  />
</form>
`;

exports[`<CustomMetricsForm /> renders update customMetric without update permision 1`] = `
<form
  onSubmit={[Function]}
>
  <DetailHeader
    text="Details"
  />
  <InputField
    componentType="input"
    disabled={true}
    id="frm-custom-metrics-name"
    inputType="text"
    label="Name *"
    name="name"
  />
  <InputField
    componentType="textarea"
    disabled={true}
    id="frm-custom-metrics-description"
    inputType="text"
    label="Description"
    name="description"
  />
  <DetailHeader
    text="Properties"
  />
  <InputField
    componentType="input"
    dataType="number"
    disabled={true}
    id="frm-custom-metrics-sla-threshold"
    inputType="text"
    label="SLA Threshold"
    labelHelpText="(seconds)"
    name="slaThreshold"
  />
  <RadioGroupField
    disabled={true}
    id="frm-custom-metrics-sla-abandon-type"
    label="SLA Abandon Type"
    name="slaAbandonType"
    options={
      Array [
        Object {
          "helpText": "(Filter Out / Exclude Short Abandons)",
          "label": "Ignore Abandons",
          "value": "ignored-abandoned-calls",
        },
        Object {
          "helpText": "(Include short abandons against/in SLA)",
          "label": "Count Against SLA",
          "value": "count-against-sla",
        },
      ]
    }
    required={true}
  />
  <InputField
    componentType="input"
    dataType="number"
    disabled={true}
    id="frm-custom-metrics-sla-abandon-threshold"
    inputType="text"
    label="Abandon Threshold"
    labelHelpText="(seconds)"
    name="slaAbandonThreshold"
  />
</form>
`;
